src/engine/Server.cpp

```diff
@@ -713,35 +714,16 @@ boost::asio::awaitable<void> Server::processQuery(
 
     // This actually processes the query and sends the result in the requested
     // format.
-    switch (mediaType.value()) {
-      using enum MediaType;
-      case csv:
-      case tsv:
-      case octetStream:
-      case sparqlXml:
-      case turtle:
-        co_await sendStreamableResponse(request, send, mediaType.value(),
-                                        plannedQuery.value(), qet,
-                                        cancellationHandle);
-        break;
-      case qleverJson:
-      case sparqlJson: {
-        // Normal case: JSON response
-        auto responseString = co_await computeInNewThread(
-            [&plannedQuery, &qet, &requestTimer, mediaType,
-             &cancellationHandle] {
-              return ExportQueryExecutionTrees::computeResultAsJSON(
-                  plannedQuery.value().parsedQuery_, qet, requestTimer,
-                  mediaType.value(), cancellationHandle);
-            },
-            cancellationHandle);
-        co_await sendJson(std::move(responseString), responseStatus);
-      } break;
-      default:
-        // This should never happen, because we have carefully restricted the
-        // subset of mediaTypes that can occur here.
-        AD_FAIL();
-    }
+    static constexpr std::array supportedTypes{
+        MediaType::csv,       MediaType::tsv,    MediaType::octetStream,
+        MediaType::sparqlXml, MediaType::turtle, MediaType::sparqlJson,
+        MediaType::qleverJson};
+    AD_CORRECTNESS_CHECK(
+        ad_utility::contains(supportedTypes, mediaType.value()));
```

Comment:
Is this check even needed?
Or can it be further inside (inside ` sendStreamableResponse` or even better inside `computeResultAsStream`, s.t. we further eliminate code from the ` Server.cpp` .